lib/swoosh/test_assertions.ex

```diff
@@ -112,6 +114,52 @@ defmodule Swoosh.TestAssertions do
     assert fun.(email)
   end
 
+  @doc ~S"""
+  Asserts multiple emails were sent.
+
+  You can pass a list of maps to match on specific params per email
+
+  ## Examples
+
+      iex> alias Swoosh.Email
+      iex> import Swoosh.TestAssertions
+
+      iex> emails = Enum.map(1..2, fn n -> Email.new(subject: "Hello, Avengers #{n}!") end)
+      iex> Swoosh.Adapters.Test.deliver_many(emails, [])
+
+      # assert a specific email was sent
+      iex> assert_emails_sent(emails)
+
+      # assert the list of emails with specific field(s) that were sent
+      iex> assert_email_sent([
+        %{subject: "Hello, Avengers 1!"},
+        %{subject: "Hello, Avengers 2!"},
+      ])
+  """
+  @spec assert_emails_sent() :: tuple | no_return
+  def assert_emails_sent do
+    assert_receive {:emails, _}
+  end
+
+  @spec assert_emails_sent([assert_email()]) ::
+          :ok | tuple | no_return
+  def assert_emails_sent([%Swoosh.Email{} | _] = emails) do
+    assert_received {:emails, ^emails}
+  end
+
+  def assert_emails_sent([%{} | _] = params_map_list) do
+    assert_received {:emails, emails}
+
+    if length(emails) != length(params_map_list),
+      do: raise("The arguments must have the same length")
```

Comment:
```suggestion
    assert length(emails) == length(params_map_list)
```

Is this better or worse? 🤔

Reply:
Good question. I'm wondering whether using `assert` could lead to confusion since it's an internal assertion in an assert (in this case `assert_emails_sent`.

Reply:
Thought "The arguments" was kinda vague